### PR TITLE
AppVeyor: Added SkipPublisherCheck to Install-Module calls for installing Pester - Fixes #140

### LINKED ISSUE #140: Unable to use AppVeyor build worker 'Visual Studio 2017' because of failing when installing Pester 4.x
When the build worker does not have WMF 5.1 the common test for module manifest will fail when a module contains class based resources.

https://ci.appveyor.com/project/PowerShell/xsqlserver/build/5.0.1215.0#L202
```
  Describing Common Tests - Module Manifest
    [+] Should contain a PowerShellVersion property of at least 5.0 based on resource types 754ms
    [-] Error occurred in Describe block 54ms
      Exception calling "ParseFile" with "3" argument(s): "Unrecoverable error in Windows PowerShell."
      At C:\projects\xsqlserver\DscResource.Tests\TestHelper.psm1:506 char:9
```

A quick fix for this is to use a build worker that contains WMF 5.1, like the image 'Visual Studio 2017'. But because Pester 3.4.0 is installed default on build worker 'Visual Studio 2017' the test run fails with the following message.

https://ci.appveyor.com/project/johlju/xsqlserver/build/5.0.603.0
```
PackageManagement\Install-Package : The version '4.0.3' of the module 'Pester' being installed is not catalog signed. Ensure that the version '4.0.3' of the module 'Pester' has the catalog file 'Pester.cat' and signed with the 
same publisher 'CN=Microsoft Root Certificate Authority 2010, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' as the previously-installed module '4.0.3' with version '3.4.0' under the directory 'C:\Program 
Files\WindowsPowerShell\Modules\Pester\3.4.0'. If you still want to install or update, use -SkipPublisherCheck parameter.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1772 char:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package], Exception
    + FullyQualifiedErrorId : ModuleIsNotCatalogSigned,Validate-ModuleAuthenticodeSignature,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage
```

So I suggest that we add the parameter `-SkipPublisherCheck` to both [rows 58 and 62](https://github.com/PowerShell/DscResource.Tests/blob/dev/AppVeyor.psm1#L58-L62)

### PR DESCRIPTION
**Pull request by johlju**
* Added SkipPublisherCheck to Install-Module calls for installing Pester.
  This way it does not conflict will signed Pester module included in Windows.

Parameter `-SkipPublisherCheck` will able to install Pester 4.0.3 for AppVeyor build image 'Visual Studio 2017'.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/141)

### CODE REVIEW

**Comment by PlagueHO** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/141#-:-Klwg_frOtvEpnWqoCxH:bnfp4nl)*

**Comment by johlju** [comment]
@kwirkykat Do you have time to look at this PR?

File: README.md [hunk 1]
```diff
@@ -345,6 +345,8 @@ Invoke-AppveyorAfterTestTask `
 * Added a common test to verify so that no markdown files contains Byte Order Mark (BOM) (issue #108).
 * Fixed bug where node_modules directory caused errors with long file paths for
   tests
+* Added SkipPublisherCheck to Install-Module calls for installing Pester.
+  This way it does not conflict will signed Pester module included in Windows.
```
Comment (kwirkykat) [inline README.md]: Typo
will -> with

Comment (johlju) [inline README.md]: Done

**Pull request merged**